sql_files/queries.sql

```diff
@@ -9,3 +9,71 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> 'Gabumon';
 SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
 
+BEGIN;
+UPDATE animals SET species = 'digimon';
+WHERE name LIKE '%mon';
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals SET species = 'pokemon'
+WHERE species IS NULL OR species = '';
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+
+BEGIN;
+DELETE FROM animals
+WHERE date_of_births > '2022-01-01';
+SAVEPOINT sp1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO sp1;
+UPDATE animals SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
+
+-- QUEIES 
+-- How many animals are there?
+SELECT COUNT(*)
+FROM animals;
+
+-- 2.How many animals have never tried to escape?
+SELECT COUNT(*)
+FROM animals
+WHERE escape_attempts = 0;
+
+-- 3.What is the average weight of animals?
+SELECT AVG(weight_kg)
+FROM animals;
+
+-- 4.Who escapes the most, neutered or not neutered animals?
+SELECT 'Neutered' AS Neutered_Status, SUM(escape_attempts) AS Total_Escape_Attempts
+FROM animals
+WHERE neutered = true
+UNION ALL
+SELECT 'Not Neutered' AS Neutered_Status, SUM(escape_attempts) AS Total_Escape_Attempts
+FROM animals
+WHERE neutered = false;
+
+-- 5.What is the minimum and maximum weight of each type of animal?
+SELECT 'Neutered' AS Neutered_Status, AVG(weight_kg) AS average_weight
+FROM animals
+WHERE neutered = true
+UNION ALL
+SELECT 'Not Neutered' AS Neutered_Status, AVG(weight_kg) AS average_weight
+FROM animals
+WHERE neutered = false;
```

Comment:
- [ ] The query above doesn't necessarily answer the question "What is the minimum and maximum weight of each type of animal?" I'd suggest using the `MIN()` and `MAX()` functions for the `weight_kg` attribute and also consider using the `species` attribute instead of categorizing them with the `neutered` attribute.